simple/provider.go

```diff
@@ -30,24 +30,29 @@ type Provider struct {
 
 // Option defines the functional option type that can be used to configure
 // provider instances
-type Option func(*Provider)
+type ProvOption func(*Provider)
+
+// ProvOptions provide a namespace for NewProvider options
+type ProvOptions struct{}
+
+var ProvOpts ProvOptions
```

Comment:
I think it's a good idea to namespace the options. I would suggest using separate packages for the Provider and Reprovider instead of using a global variable for the options (because if we use a global variable it doesn't really make sense to use variadic options).

So we would have a `SimpleProvider` in a `provider` package and a `SimpleReprovider` in a `reprovider` package

Reply:
that was my original implementation (see https://github.com/ipfs/go-ipfs-provider/tree/7ecc0ec5d94e3ed3650fdb261c4b13c33c9595df) but it doesn't fit together so well because the main package with the interfaces is also named `provider`.

Reply:
I see. I think it's ok, maybe give the `provider` package a different name

Reply:
what about `provider` for the root package (interfaces, System, offlineProvider), `sprov` for the SimpleProvider and `sreprov` for the SimpleReprovider ?

Reply:
SGTM 👍